### PR TITLE
Feature/auth flow

### PR DESCRIPTION
**Pull request by alokesh985**
Created a UI, with validations in the input fields, and floating labels.
Screenshot attached.
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +25,8 @@
     "redux-logger": "^3.0.6",
     "redux-persist": "^6.0.0",
     "redux-saga": "^1.1.3",
-    "typescript": "~3.7.2"
+    "typescript": "^4.1.6",
+    "yarn": "^1.22.11"
```
Comment (aryankush25) [inline package.json]: @alokesh985 Remove this and add it globally to your machine.

File: src/container/AppBoot/index.tsx [hunk 2]
```diff
@@ -13,14 +13,17 @@ import 'react-toastify/dist/ReactToastify.css';
...
         <div className="app-container">
+          {/* Used to preserve previous component path */}
           <ConnectedRouter history={history}>
             <AppRoutes />
           </ConnectedRouter>
         </div>
 
-        <ToastContainer
+        {/* react-toastify is used to add notifications */}
+        {/* <ToastContainer
```
Comment (aryankush25) [inline src/container/AppBoot/index.tsx]: @alokesh985 Why is this ToastContainer commented?

File: src/container/AuthContainer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import logo from './../../assets/images/bookmarks-app-logo.png';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import './styles.scss';
+
+const AuthContainer = () => {
+  type Inputs = {
+    'User Name': string;
```
Comment (aryankush25) [inline src/container/AuthContainer/index.tsx]: @alokesh985 Change it to email.

File: src/global.d.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+declare module '*.scss' {
```
Comment (aryankush25) [inline src/global.d.ts]: @alokesh985 Can you explain this?

File: src/routes/Login/index.tsx [hunk 5]
```diff
@@ -1,14 +1,114 @@
-import React from 'react';
-import { useLoginHook } from './hooks';
-import SharedButton from '../../components/shared/SharedButton';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { SIGNUP_ROUTE } from './../../utils/routesConstants';
+import './Login.scss';
```
Comment (aryankush25) [inline src/routes/Login/index.tsx]: @alokesh985 Change this file name to `./login.scss`.

File: src/routes/Login/index.tsx [hunk 6]
```diff
@@ -1,14 +1,114 @@
...
+        <img src={logo} alt="Image Not Available"></img>
+      </div>
+
+      <div className="fields-container">
+        <div className="auth">
+          <h1 className="main-welcome">Welcome back!</h1>
+          <h3 className="subText">Log in to your account.</h3>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <div className="input-container">
+              <input
```
Comment (aryankush25) [inline src/routes/Login/index.tsx]: @alokesh985 Make a reusable for this input in components/shared.

File: src/routes/Login/index.tsx [hunk 7]
```diff
@@ -1,14 +1,114 @@
...
+                  EmailRegister.onChange(e);
+                  handleChangeEmail(e);
+                }}
+              />
+              <label className={EmailValue && 'filled'}>Email ID</label>
+            </div>
+            {errors.EmailID && <p>{errors.EmailID.message} </p>}
+
+            <div className="input-container">
+              <input
```
Comment (aryankush25) [inline src/routes/Login/index.tsx]: @alokesh985 Make a reusable for this input in components/shared.

File: src/routes/Signup/index.tsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+  return (
+    <div className="signup-wrapper">
+      <div className="image">
+        <img src={logo} alt="Image Not Available"></img>
+      </div>
+      <div className="fields-container">
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <div className="input-container">
+            <input
```
Comment (aryankush25) [inline src/routes/Signup/index.tsx]: @alokesh985 Make a reusable for this input in components/shared.

File: src/routes/Signup/index.tsx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+                EmailRegister.onChange(e);
+                handleChangeEmail(e);
+              }}
+            />
+            <label className={EmailID && 'filled'}>Email ID</label>
+          </div>
+          {errors['User Name'] && <p>Email ID Field is required</p>}
+          {errors.EmailID && <p>{errors.EmailID.message} </p>}
+          <div className="input-container">
+            <input
```
Comment (aryankush25) [inline src/routes/Signup/index.tsx]: @alokesh985 Make a reusable for this input in components/shared.

File: src/routes/Signup/Signup.scss [hunk 10]
```diff
@@ -0,0 +1,214 @@
+@import url('https://fonts.googleapis.com/css2?family=Poppins');
```
Comment (aryankush25) [inline src/routes/Signup/Signup.scss]: @alokesh985 What is the need to add this here?

File: src/routes/Signup/Signup.scss [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+
+$header-font-stack: Poppins, sans-serif;
```
Comment (aryankush25) [inline src/routes/Signup/Signup.scss]: @alokesh985 Move it to global file.

**Review by aryankush25 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (alokesh985) [reply package.json]: Done.

File: src/container/AppBoot/index.tsx (hunk 2 above)
Comment (alokesh985) [reply src/container/AppBoot/index.tsx]: I had commented it out for testing something. Uncommented now.

File: src/container/AuthContainer/index.tsx (hunk 3 above)
Comment (alokesh985) [reply src/container/AuthContainer/index.tsx]: Done.

File: src/global.d.ts (hunk 4 above)
Comment (alokesh985) [reply src/global.d.ts]: image

I was having some difficult while importing .scss files into Typescript. So, I looked it up online. I found this.
However, now, it seems to be working even after removing this file.

File: src/routes/Login/index.tsx (hunk 5 above)
Comment (alokesh985) [reply src/routes/Login/index.tsx]: Done.

File: src/routes/Signup/Signup.scss (hunk 10 above)
Comment (alokesh985) [reply src/routes/Signup/Signup.scss]: The Poppins font was not loading initially.
Now, it's working fine. Removing from here. Adding to global file.

File: src/routes/Signup/Signup.scss (hunk 11 above)
Comment (alokesh985) [reply src/routes/Signup/Signup.scss]: Done.

File: src/routes/Signup/index.tsx (hunk 9 above)
Comment (alokesh985) [reply src/routes/Signup/index.tsx]: Done.

**Comment by alokesh985** [comment]
Made the changes. Kindly please re-check

**Pull request merged**